src/rp2_common/hardware_gpio/include/hardware/gpio.h

```diff
@@ -365,46 +375,243 @@ enum gpio_drive_strength gpio_get_drive_strength(uint gpio);
  *   1 | High level
  *   2 | Edge low
  *   3 | Edge high
+ *
+ *  Which are specified in \ref gpio_irq_level
  */
 void gpio_set_irq_enabled(uint gpio, uint32_t events, bool enabled);
 
-/*! \brief Enable interrupts for specified GPIO
+#ifndef GPIO_CALLBACK_IRQ_ORDER_PRIORITY
+#define GPIO_CALLBACK_IRQ_ORDER_PRIORITY PICO_SHARED_IRQ_HANDLER_LOWEST_ORDER_PRIORITY
+#endif
+
+/*! \brief Set the generic callback used for GPIO IRQs events for the current core
  *  \ingroup hardware_gpio
  *
- * \note The IO IRQs are independent per-processor. This configures IRQs for
+ * This function sets the callback used for all GPIO IRQs on the current core that are not explicitly
+ * hooked via \ref gpio_add_raw_irq_handler or other gpio_add_raw_irq_handler_ functions.
+ *
+ * This function is called with the GPIO number and event mask for each of the (not explicitly hooked)
+ * GPIOs that have events enabled and that are pending (see \ref gpio_get_irq_events).
+ *
+ * \note The IO IRQs are independent per-processor. This function affects
  * the processor that calls the function.
  *
- * \param gpio GPIO number
- * \param events Which events will cause an interrupt. See \ref gpio_set_irq_enabled for details.
- * \param enabled Enable or disable flag
- * \param callback user function to call on GPIO irq. Note only one of these can be set per processor.
+ * \param callback default user function to call on GPIO irq. Note only one of these can be set per processor.
+ */
+void gpio_set_irq_callback(gpio_irq_callback_t callback);
+
+/*! \brief Convenience function which performs multiple GPIO IRQ related initializations
+ *  \ingroup hardware_gpio
+ *
+ * This method is a slightly eclectic mix of initialization, that:
+ *
+ * \li Updates whether the specified events for the specified GPIOs cause an interrupt on the calling core bease
```

Comment:
This function takes a single `uint gpio`, so should "specified GPIOs" be "specified GPIO" ?